packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -197,12 +214,73 @@ export function getDevMiniflareOptions(
 	viteDevServer: vite.ViteDevServer
 ): MiniflareOptions {
 	const resolvedViteConfig = viteDevServer.config;
+	const logger = new ViteMiniflareLogger(resolvedViteConfig);
+
 	const entryWorkerConfig = getEntryWorkerConfig(resolvedPluginConfig);
 	const assetsConfig =
 		resolvedPluginConfig.type === "assets-only"
 			? resolvedPluginConfig.config.assets
 			: entryWorkerConfig?.assets;
 
+	const headersFile = path.join(resolvedViteConfig.publicDir, HEADERS_FILENAME);
+	const redirectsFile = path.join(
+		resolvedViteConfig.publicDir,
+		REDIRECTS_FILENAME
+	);
```

Comment:
I think the `_headers` and `_redirects` files should be in the project root rather than the `public` directory. The `public` directory is only a small part of assets handling in Vite. They should then be emitted as part of the client bundle at build time but we can handle that in a separate PR.

```suggestion
	const headersFile = path.join(resolvedViteConfig.root, HEADERS_FILENAME);
	const redirectsFile = path.join(
		resolvedViteConfig.root,
		REDIRECTS_FILENAME
	);
```

Reply:
Not sure I agree on that. Most frameworks (e.g. [Astro](https://docs.astro.build/en/guides/integrations-guide/cloudflare/#headers)) expect users to author these in a public directory.

Reply:
Yeah I think we have gone with these staying in the public directory for now. As tested in https://github.com/cloudflare/workers-sdk/pull/8496.